graspologic/nominate/VNviaSGM.py

```diff
@@ -0,0 +1,303 @@
+            msg = "Adjacency matrix entries must be square"
+            raise ValueError(msg)
+
+        if len(seeds) == 0:
+            print("Must include at least one seed to produce nomination list")
+            return None
+        if len(seeds) != 2:
+            msg = "List must be length two, with first element containing seeds \
+                  of A and the second containing seeds of B"
+            raise ValueError(msg)
+
+        seedsA = seeds[0]
+        seedsB = seeds[1]
+
+        if len(seedsA) != len(seedsB):
+            msg = "Must have the same number of seeds for each adjacency matrix"
+            raise ValueError(msg)
+        if len(seedsA) == 0:
+            print("Must include at least one seed to produce nomination list")
+            return None
+
+        voi = np.reshape(np.array(voi), (1,))
+
+        # get reordering for Ax
+        nsx1 = np.setdiff1d(np.arange(A.shape[0]), np.concatenate((seedsA, voi)))
+        a_reord = np.concatenate((seedsA, voi, nsx1))
+
+        # get reordering for B
+        nsx2 = np.setdiff1d(np.arange(B.shape[0]), seedsB)
+        b_reord = np.concatenate((seedsB, nsx2))
+
+        AA = A[np.ix_(a_reord, a_reord)]
+        BB = B[np.ix_(b_reord, b_reord)]
+
+        seeds_reord = np.arange(len(seedsA))
+        voi_reord = len(seedsA)
+
+        subgraph_AA = np.array(
+            _get_induced_subgraph_list(
+                AA, self.order_voi_subgraph, voi_reord, mindist=1
+            )
+        )
+
+        voi_reord = np.reshape(np.array(voi_reord), (1,))
+
+        Sx1 = np.intersect1d(subgraph_AA, seeds_reord)
+
+        if len(Sx1) <= 0:
+            print(
+                "Voi {} was not a member of the induced subgraph A[{}]".format(
+                    voi, seedsA
+                )
+            )
+            return None
+
+        Nx1 = np.array(
+            _get_induced_subgraph_list(
+                AA, self.order_seeds_subgraph, list(Sx1), mindist=0
+            )
+        )
+        Nx2 = np.array(
+            _get_induced_subgraph_list(
+                BB, self.order_seeds_subgraph, list(Sx1), mindist=0
+            )
+        )
+
+        foo = np.concatenate((Sx1, voi_reord))
+        ind1 = np.concatenate((Sx1, voi_reord, np.setdiff1d(Nx1, foo)))
+        ind2 = np.concatenate((Sx1, np.setdiff1d(Nx2, Sx1)))
+
+        AA_fin = AA[np.ix_(ind1, ind1)]
+        BB_fin = BB[np.ix_(ind2, ind2)]
+
+        seeds_fin = list(range(len(Sx1)))
+        sgm = GMP(n_init=self.n_init, shuffle_input=False, init="rand", padding="naive")
+        corr = sgm.fit_predict(AA_fin, BB_fin, seeds_A=seeds_fin, seeds_B=seeds_fin)
+        P_outp = sgm.probability_matrix_
+
+        b_inds = b_reord[ind2]
+        self.n_seeds = len(Sx1)
+
+        nomination_list = list(zip(b_inds[self.n_seeds :], P_outp[0]))
+        nomination_list.sort(key=lambda x: x[1], reverse=True)
+        self.nomination_list = np.array(nomination_list)
+        return self
+
```

Comment:
can you comment this section so it's more clear what you're doing

Reply:
This section has been commented. Let me know if you want more/less detail.